armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -184,6 +185,30 @@ def construct(
                         # learn mult from grid definition
                         c.setDimension("mult", len(c.spatialLocator))
 
+                idsInGrid = list(gridDesign.gridContents.values())
+                if componentDesign.latticeIDs:
+                    for latticeID in componentDesign.latticeIDs:
+                        allLatticeIds.add(str(latticeID))
+                        # the user has given this component latticeIDs. check that
+                        # each of the ids appears in the grid, otherwise
+                        # their blueprints are probably wrong
+                        if len([i for i in idsInGrid if i == str(latticeID)]) == 0:
+                            raise ValueError(
+                                f"latticeID {latticeID} in block blueprint '{self.name}' is expected "
+                                "to be present in the associated block grid. "
+                                "Check that the component's latticeIDs align with the block's grid."
+                            )
```

Comment:
Just a thought.

This method is suddenly quite long.

Could we move this to a private helper method that just does the error handling? What do you think?